### ISSUE TITLE
Error in running run_tf_text_classification.py

### ISSUE THREAD

**Issue by rajesh-dhiman** [huggingface/transformers#11171]
## Environment info

- `transformers` version: 4.4.2
- Platform: Windows-10-10.0.18362-SP0
- Python version: 3.8.0
- PyTorch version (GPU?): 1.7.1+cpu (False)
- Tensorflow version (GPU?): 2.4.1 (False)
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

## Information

Model I am using (Bert, XLNet ...):  microsoft/deberta-base

The problem arises when using: run_tf_text_classification.py

## To reproduce

just run
python run_tf_text_classification.py --model_name_or_path microsoft/deberta-base --output_dir classificationoutput --train_file PreparedData.csv --label_column_id 1 --do_train

[PreparedData.zip](https://github.com/huggingface/transformers/files/6288064/PreparedData.zip)

**Comment by rajesh-dhiman** [comment]
@Rocketknight1 can you help in this

**Comment by rajesh-dhiman** [comment]
@sgugger

**Comment by LysandreJik** [comment]
Hi, we have a new, simpler and more robust text classification in TensorFlow script contributed by @Rocketknight1 [here](https://github.com/huggingface/transformers/blob/master/examples/tensorflow/text-classification/run_text_classification.py), could you check it out?

**Comment by rajesh-dhiman** [comment]
Hi @LysandreJik   and @Rocketknight1 

I used it   run_text_classification.py in jupyter
%run run_text_classification.py \
--model_name_or_path roberta-base \
--output_dir classificationoutput \
--train_file PreparedData.csv \
--validation_file PreparedData.csv \
--do_train


PreparedData.csv looks like below
sentence,label
sent1,l1
sent2,l1
sent3,l2
sent3,l2

I got following error

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~\run_text_classification.py in <module>
    532 
    533 if __name__ == "__main__":
--> 534     main()

~\run_text_classification.py in main()
    492 
    493         callbacks = [SavePretrainedCallback(output_dir=training_args.output_dir)]
--> 494         model.fit(
    495             training_dataset, validation_data=eval_dataset, epochs=training_args.num_train_epochs, callbacks=callbacks
    496         )

c:\python38\lib\site-packages\tensorflow\python\keras\engine\training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq, max_queue_size, workers, use_multiprocessing)
   1086           self._maybe_load_initial_epoch_from_ckpt(initial_epoch))
   1087       logs = None
-> 1088       for epoch, iterator in data_handler.enumerate_epochs():
   1089         self.reset_metrics()
   1090         callbacks.on_epoch_begin(epoch)

c:\python38\lib\site-packages\tensorflow\python\keras\engine\data_adapter.py in enumerate_epochs(self)
   1132     with self._truncate_execution_to_epoch():
   1133       data_iterator = iter(self._dataset)
-> 1134       for epoch in range(self._initial_epoch, self._epochs):
   1135         if self._insufficient_data:  # Set by `catch_stop_iteration`.
   1136           break

TypeError: 'float' object cannot be interpreted as an integer

​

**Comment by Rocketknight1** [comment]
Good catch, thank you! This was totally my fault, and has now been fixed in #11379 . If you pull the latest version of the library, training should work.
